docs/contribute/plugin/intro.md

```diff
@@ -4,34 +4,99 @@ sidebar_position: 1
 
 # WasmEdge Plug-in System Introduction
 
-While the WasmEdge language SDKs allow registering host functions from a host (wrapping) application, developers should implement the host functions before compilation.
+While the WasmEdge language SDKs allow registering host functions from a host (wrapping) application, developers should implement the host functions before compilation. However, for a more flexible and dynamic extension of the host functions, WasmEdge provides a plug-in architecture to load the plug-in shared library.
 
-For the other solutions, WasmEdge provides the plug-in architecture to load the plug-in shared library for easier extending of the host functions.
+A WasmEdge plug-in is a software component that extends the functionality of the WasmEdge runtime. Currently, developers can follow the guides to implement the plug-ins in [C API](develop_plugin_c.md) (recommended) or [C++](develop_plugin_cpp.md). With the help of the WasmEdge SDKs in the supported languages, developers can load and register the host functions from the plug-in shared libraries, allowing them to seamlessly integrate the plug-ins into the WasmEdge runtime as if they were part of the core runtime.
 
-With developing the plug-ins, WasmEdge SDKs in the supported languages can load and register the host functions from the plug-in shared libraries.
+```mermaid
+graph LR
+    A((Host Application)) -- Loads --> B((Plugin Shared Library))
+    B -- Registers --> C((Wasmedge Runtime))
+```
 
-In current, developers can follow the guides to implement the plug-ins in [C API (recommended)](develop_plugin_c.md) or [C++](develop_plugin_cpp.md).
+In this diagram, the _Host Application_ represents the application or environment where the Wasmedge runtime is embedded or used. The _Plug-in Shared Library_ refers to the library containing the plug-in code and functions that extend the functionality of the Wasmedge runtime. The _Wasmedge Runtime_ represents the runtime environment that executes WebAssembly modules, including the core runtime and any registered plug-ins.
+
+## Usages of WasmEdge Plug-in
+
```

Comment:
This section is not correct.
The plug-in of WasmEdge only provides the portable host modules and host functions for WASM extension. We don't have such cases you mentioned yet.

Reply:
Still not correct. Delete this section.

Reply:
PLEASE DO NOT RESOLVE THIS WITHOUT REQUESTED MODIFICATION.